core/js/src/typespecs/app_types.ts

```diff
@@ -0,0 +1,362 @@
+// Type definitions for operating on the app database.
```

Comment:
The stuff in `typespecs` can be ignored, as we are just moving the files unmodified from the internal repo.